packages/repeater/src/request-runner/Request.ts

```diff
@@ -73,16 +113,17 @@ export class Request {
       ...headers
     };
 
-    this._headers = Object.fromEntries(
-      Object.entries(mergedHeaders).map(
-        ([field, value]: [string, string | string[]]) => [
-          field,
+    this._headers = Object.entries(mergedHeaders).reduce(
```

Comment:
Unclear why di you make such changes. Revert them; we do not use reduce to create objects when it is possible to do so from entries.